### PR TITLE
Release 2024.3

### PR DESCRIPTION
**Pull request by josephbirkner**
The WASM `Feature` now has an `inspectionModel` method.

Also, this PR fixes a potential NaN error in `localWgs84UnitCoordinateSystem`.

### CODE REVIEW

File: libs/core/src/inspection.cpp [hunk 1]
```diff
@@ -0,0 +1,339 @@
...
+        auto singleValueForField = convertField(kk, v);
+        if (singleValueForField) {
+            ++numValues;
+            singleValue = singleValueForField;
+        }
+        ++index;
+    }
+
+    if (numValues == 1)
+        return singleValue;
```
Comment (josephbirkner) [inline libs/core/src/inspection.cpp:264]: Need to assign singleValue to fieldScope here.

**Comment by Waguramu** [comment]
- [ ]  Add color scheme for value types in the inspection panel (need input from Serein).

**Comment by josephbirkner** [comment]
- [ ] Fix bug with disappearing marker on jump
- [ ] Fix code duplication for copy to clipboard and the marker graphics.

**Comment by MisterGC** [comment]
My feedback after a longer testing session - let's discuss: 
 
### Feature Search

- [ ]  **Result List Size:** Run a search with many results, scroll to the bottom, then run a search with few results -> list is too small.
- [ ]  **Clearing Search Results:** Why does changing to the Data/Style selection clear the search result?
- [ ]  **Terminology 'Match Feature':** Used term 'Match Feature' - don't we talk about feature search? Maybe we should also give an indication about the scope (loaded/visible tiles!?).
- [ ]  **Activate Search via Enter:** Although we plan to refine search option selection via keyboard later - we could at least enable that pressing enter selects the first match in the list - e.g., feature search.
- [ ]  **Search/Jump Actions List:** The list of search/jump actions is longer than the result list. I'd expect the result list is always there, and the search/jump targets appear as a smaller list when entering the search field. Currently, the user has no option to bring back the current result list, only re-triggering the search leads to a new list.
- [ ] **Map Label in Results:** The list of results no longer contains the map label. Do we need it in the future? I'd leave it out for now, as soon as the inspection shows also the map ID the state should/could be acceptable.

### Jump To

- [ ]  **Too Many Entries:** When entering a jump-to request, there are too many entries of potential jump-to targets. E.g., there are three different entries for 'Jump To Lane' which are always visible - jumping to a WGS84 location needs scrolling.

### Inspection Panel

- [ ]  **Copying Entries:** Copying entries from the inspection panel is unnecessarily hard. At least on macOS with Firefox, I had to use right-click directly and was not able to mark what I wanted to copy.
- [ ]  **Filter Icon Display:** What about showing the filter option icon only if the user hovers/enters the filter field (on the right)? We could then also add a clear filter field.

### Maps And Styles

- [ ] **Context Menu for Options:** Why not use a context menu regarding the "Check/Uncheck all ..." options? The current indicators look more like the entries would be expandable.
- [ ]  **Migration of 'Reload' and 'Export':** Do you still plan to migrate the 'Reload from Disc' and 'Export Style' to the style editor to make the interface even less cluttered?

### Location Marker

- [ ]  **Marker Tool Icon:** I'd expect the icon to change the other way around. When the marker is active, it is not crossed out. If not, it is - I'd also use grey color in addition to the crossing to make it quite obvious.
- [ ]  **Decoupling Marker and Coordinates:** We should also decouple marker toggling from displaying the coordinates. What about showing the coordinates if the marker tool is not activated?

**Comment by MisterGC** [comment]
And some more:

### Feature Search
- [ ] I zoomed into a  map searched for some criteria e.g. `*.isArtificial == true`, all hits are shown, then I zoom out and want to apply the search on the viewport, but retriggering the search seem to have no effect, only if I change the search term and run it, a new search is triggered
      cannot_rerun_search_without_changing_term
- [ ] if you run (multiple) searches with a lot of hits, the progress bar behaves in an unexpected way
       progress_bar_unexpected_label_n_progress

### Attribute Inspection
- [ ]  the attribute panel applies now the filter term after selecting a new element, but if I pick a non-feature position in between and then a feature again, the inspection still applies the filter although the filter field is empty
- [ ]

**Comment by MisterGC** [comment]
### Feature Search
- [ ] For upcoming iterations - create corresponding issue - when I search for landmarks and I get some hit with a geometry perpendicular to earth surface - I can barely see the feature when I click on the result list
      landmark_as_result_confusion

**Review by MisterGC — APPROVED** [review]
Approve this, I do. Good work, your effort shows.

**Pull request merged**